### PR TITLE
DD4hep: DDTIDAxialCable Algorithm

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

Port of a DDTIDAxialCable algorithm to DD4hep

@vargasa - FYI, the numbers in xml are kind of random. Please, feel free to modify them.

#### PR validation:

```
cmsRun DetectorDescription/DDCMS/test/python/testDDTIDAxialCableAlgorithm.py
```

#### if this PR is a backport please specify the original PR:

no backport is needed
Screen Shot 2019-05-03 at 12 55 44

### CODE REVIEW

**Comment by ianna** [comment]
@smuzaffar - I think anything after a ```{``` should be indented to avoid this:
```
}
} // namespace
```

**Comment by ianna** [comment]
@smuzaffar - I don't think this formatting is particularly useful:
```
-    LogVerbatim("TIDGeom") << "DDTIDAxialCableAlgo test: "
-                        << solid.name()
-                        << " Polycone made of "<< matIn
-                        << " from " << convertRadToDeg(-0.5*width)
-                        << " to " << convertRadToDeg(0.5*width)
-                        << " and with "  << pconZ.size() << " sections ";
-
+    LogVerbatim("TIDGeom") << "DDTIDAxialCableAlgo test: " << solid.name() << " Polycone made of " << matIn << " from "
+                           << convertRadToDeg(-0.5 * width) << " to " << convertRadToDeg(0.5 * width) << " and with "
+                           << pconZ.size() << " sections ";

```
as you can see, the lines are too long.

**Comment by ianna** [comment]
@smuzaffar - will the code checks be re-triggered? There was a commit just after it started.

**Comment by smuzaffar** [comment]
look like bot thinks that it has already processed your last commit. I can re-trigger it but you can do it to by adding `code-checks` comment

**Comment by smuzaffar** [comment]
code-checks

**Comment by ianna** [comment]
@smuzaffar - ok, it looks like after ```scram build code-checks``` the modified by it file was excluded from ```scram build code-format```

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

File: Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+  Volume mother = ns.volume(args.parentName());
+
+  LogVerbatim("TIDGeom") << "DDTIDAxialCableAlgo debug: Parent "
+                         << mother.name() << "\tChild " << childName
+                         << " NameSpace " << ns.name() << "\tMaterial " << matIn
+                         << " and " << matOut;
+
+  std::vector<Volume> logs;
+  double thk = thick / zposRing.size();
```
Comment (cvuosalo) [inline Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc]:
```suggestion
  const double thk = thick / zposRing.size();
```

File: Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                         << " Tubs made of " << matOut << " from "
+                         << convertRadToDeg(-0.5 * width) << " to "
+                         << convertRadToDeg(0.5 * width) << " with Rin " << r
+                         << " Rout " << rTop << " ZHalf "
+                         << 0.5 * (zEnd - zBend);
+
+  logs.emplace_back(Volume(solid.name(), solid, ns.material(matOut)));
+
+  // Position the cables
+  double theta = 90_deg;
```
Comment (cvuosalo) [inline Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc]:
```suggestion
  constexpr double theta = 90_deg;
```

Comment (fabiocos) [reply Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc]: @ianna @cvuosalo do you agree in updating this PR?

**Comment by ianna** [comment]
please test

File: Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc (hunk 2 above)
Comment (cvuosalo) [reply Geometry/TrackerCommonData/plugins/dd4hep/DDTIDAxialCableAlgo.cc]: @fabiocos: Looks good to me.

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**